engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/Boolean.java

```diff
@@ -10,7 +10,11 @@
 // Before moving this definition to the `bool` package, as we should, one would have to address that
 // problem first.
 @BuiltinType(name = "Standard.Base.Data.Boolean.Boolean")
-public class Boolean extends Builtin {
+public final class Boolean extends Builtin {
+  public Boolean() {
+    super(java.lang.Boolean.class);
```

Comment:
Now we attempt to provide a `representationType` to each `Builtin`. That way we can deduce from signature methods having argument like `boolean` that the desired builtin is `Boolean` and the Enso type should be `Standard.Base.Data.Boolean.Boolean`. With that information we can extract (via `CastToNode`) such a type from `EnsoMultiValue`....